ONSdigital/response-operations-ui
# Pass classifiers in create survey call

# Motivation and Context

In a step toward being able to support creation of classifiers alongside creating a survey, the survey service now takes a set of desired classifiers in its create call. This means this service needs to add those to the call survey service.

# What has changed

The `POST` call to the survey service now adds an extra bit of `json` to the payload to specify the classifiers to be created. For now it hardcodes to the "business" set of classifiers.
Future enhancement is to make this configurable based on user input.

```python
survey_details = {
    "surveyRef": <survey_ref>,
    "shortName": <short_name>,
    "longName": <long_name>,
    "legalBasisRef": <legal_basis>,
    "surveyType": "Business",
    "classifiers": [
        {"name": "COLLECTION_INSTRUMENT", "classifierTypes": ["FORM_TYPE"]},
        {"name": "COMMUNICATION_TEMPLATE", "classifierTypes": ["LEGAL_BASIS", "REGION"]}
    ]
}
```

# How to test?

> **Important** This PR is reliant on functionality of the survey service introduced in https://github.com/ONSdigital/rm-survey-service/pull/70

- Run acceptance tests
- Create a survey via the `/surveys/create` page - check the survey database to see that the classifiers have been created.


# Links

- Survey service PR https://github.com/ONSdigital/rm-survey-service/pull/70

# Screenshots (if appropriate):

n/a
